aiida/cmdline/params/options/interactive.py

```diff
@@ -44,30 +44,42 @@ def foo(label):
             click.echo('Labeling with label: {}'.format(label))
     """
 
-    def __init__(self, param_decls=None, switch=None, empty_ok=False, **kwargs):
+    def __init__(
+            self,
+            param_decls=None,
+            switch=None,  #
+            prompt_fn=None,
+            # empty_ok=False,
+            **kwargs):
         """
         :param param_decls: relayed to :class:`click.Option`
-        :param switch: sequence of parameter names
+        :param switch: sequence of parameter
+        :param prompt_fn: a callable that defines if the option should be asked for or not in interactive mode
         """
-        # intercept prompt kwarg
+
+        # intercept prompt kwarg; I need to pop it before calling super
         self._prompt = kwargs.pop('prompt', None)
-        if kwargs.get('required', None):
-            required_fn = kwargs.get('required_fn', lambda ctx: True)
-            kwargs['required_fn'] = lambda ctx: noninteractive(ctx) and required_fn(ctx)
 
-        # super
+        # call super
```

Comment:
If important to explain why we call super and why here than maybe we should write that as a comment. As it stands it is not very informative and I would remove it